src/dbnode/storage/series/series.go

```diff
@@ -159,7 +172,7 @@ func (s *dbSeries) updateBlocksWithLock(blockStates map[xtime.UnixNano]BlockStat
 	)
 	for startNano, currBlock := range s.cachedBlocks.AllBlocks() {
 		start := startNano.ToTime()
-		if start.Before(expireCutoff) {
+		if start.Before(expireCutoff) || timeExistsInSlice(start, evictedBucketTimes) {
```

Comment:
Do you think this will be better / faster than just doing two loops?

```golang
for _, t := range evictedBucketTimes {
    s.cachedBlocks.RemoveBlockAt(t)
}

for startNano, currBlock := range s.cachedBlocks.AllBlocks() {
    ...
}
```

Reply:
I thought about that, and while I think the performance difference is probably negligible, I preferred not refactor out the logic within that scope (including the closing the block and updating the result).